pkg/data_structure/set.go

```diff
@@ -0,0 +1,26 @@
+package data_structure
```

Comment:
data_structure is not a good name. for example in std lib, use [container/lis](https://pkg.go.dev/container/list).

for this package, the `IdentifierSet` only include small code, copy the code is better.

Reply:
If we will maintain various types of data structures, it is recommended to:
data_structure/collect/...
data_structure/graph/...
data_structure/tree/...

Reply:
This implementation is too simple..  and there is no documentation whatsoever. Many details are not clear, for example, is it thread safe... I wonder if we need to write such a toy? And there are already plenty of go set implementations available.

Reply:
removed the exported IdentifierSet data structure and use an unexported simple stringSet struct within the package `list` instead. ptal